doc/example/customization.cpp

```diff
@@ -86,8 +87,7 @@ struct near_constraint
     template< typename Actual >
     bool operator()( Actual actual ) const
     {
-        return std::abs( actual - boost::unwrap_ref( expected_ ) )
-            < boost::unwrap_ref( threshold_ );
+        return std::abs( actual - expected_ ) < threshold_ ;
```

Comment:
Tbh I don't understand that part of the docs. Imo the important part is to use (c) ref to pass things as references instead of values. The use and even existence of unwrap_ref is not required and I'd rather remove it from the docs. A cref is simply a copy able reference

Reply:
It seems it is still required. Just not here. Example: Comparing an actual `const char*` against an expected `cref(std::string)` requires converting both sides: the cref to a string and the pointer to a string. This doesn't work (only 1 conversion allowed). However here it is only int vs cref(int) which works fine. even int vs cref(float) works.

Added a new test case to showcase that

I added it back here.

Reply:
Nice, thanks!